app/src/main/java/com/nutomic/syncthingandroid/fragments/StatusFragment.java

```diff
@@ -0,0 +1,260 @@
+            Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_status, container, false);
+    }
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        mAdapter = new ArrayAdapter(getActivity(), android.R.layout.simple_list_item_1);
+        setListAdapter(mAdapter);
+        setHasOptionsMenu(true);
+        updateStatus();
+    }
+
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        mActivity = (MainActivity) getActivity();
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        inflater.inflate(R.menu.status_list, menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.open_preferences:
+                startActivity(new Intent(getContext(), SettingsActivity.class));
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    private void updateStatus() {
+        SyncthingActivity syncthingActivity = (SyncthingActivity) getActivity();
+        if (syncthingActivity == null || getView() == null || syncthingActivity.isFinishing()) {
+            return;
+        }
+        SyncthingService syncthingService = syncthingActivity.getService();
+        if (syncthingService == null) {
+            return;
+        }
+
+        // Add status line showing the syncthing service state.
+        ArrayList<String> statusItems = new ArrayList<String>();
+        switch (mServiceState) {
+            case INIT:
+            case STARTING:
+                statusItems.add(getString(R.string.syncthing_starting));
+                break;
+            case ACTIVE:
+                statusItems.add(getString(R.string.syncthing_running));
+                break;
+            case DISABLED:
+                statusItems.add(getString(R.string.syncthing_not_running));
+                break;
+            case ERROR:
+                statusItems.add(getString(R.string.syncthing_has_crashed));
+                break;
+        }
+
+        // Add explanation why syncthing is (not) running.
+        switch (mServiceState) {
+            case ACTIVE:
+            case DISABLED:
+                statusItems.add(getString(R.string.reason) + "\n" +
+                    "- " + syncthingService.getRunDecisionExplanation().trim().replace("\n", "\n- "));
+            default:
+                break;
+        }
+
+        // Add status holders refreshed by callbacks to the list.
+        if (mServiceState == SyncthingService.State.ACTIVE) {
+            synchronized (mStatusHolderLock) {
+                if (!TextUtils.isEmpty(mCpuUsage)) {
+                    statusItems.add(getString(R.string.cpu_usage) + ": " + mCpuUsage);
```

Comment:
You shouldn't do `+` on strings because that won't work well with translations in various languages. Use string parameters instead.

Reply:
Or handle it in the layout with seperate TextViews.